test/unit/sql/enum.test.js

```diff
@@ -24,6 +24,26 @@ describe(Support.getTestDialectTeaser('SQL'), () => {
         }
       });
 
+      const NamedEnumUser = current.define('user', {
+        mood: {
+          type: DataTypes.ENUM({
+            name: 'mood',
+            values: ['happy', 'sad']
+          })
+        }
+      });
```

Comment:
Should there be a test case for an `ARRAY` of named `ENUM`s?

Reply:
Happy to add a test, but I don't quite follow.  Can you provide an example of exactly what you mean?

Reply:
Sorry for the late reply. Something like:

```
const NamedEnumUser = current.define('user', {
  mood: {
    type: DataTypes.ARRAY(DataTypes.ENUM({
      name: 'mood',
      values: ['happy', 'sad']
    }))
  }
});
```

To ensure an `ARRAY`s of named enums works.

Reply:
Thank you, I will test for this.

Reply:
I wrote the test for this and realized how this still does not seem right.  I'm not sure of a case where you'd want an array of enums for the type.  Apologies, maybe this is over my head but as far as I know, when you have a column in PG that is of type enum...there is only one enum type that it is tied to.  Can you help me understand a bit more?

Reply:
I think they're talking about a scenario like this:
```sql
CREATE TABLE person (
    name text,
    current_moods mood[]
);
```

...where the `mood[]` type allows the person to have multiple moods simultaneously.

Reply:
Ahh, right.  That makes sense.  Thank you.

Reply:
Thank _you_ for working on this PR! 👏

Reply:
As @nmchaves said.

An enumerated type in PostgreSQL is typically a `text` type. So think of it as me suggesting an `array` of `text` which is perfectly valid. The only difference by using an enum instead of `text` is obviously that the array values can only be one of the defined constants.